### PR TITLE
Added test for HomeScreenFragment

### LINKED ISSUE #2583: [Code health] HomeFragment Test cases
Write HomeFragment Test cases

### PR DESCRIPTION
**Pull request by anandwana001**
Fixes #2583 

- This is a part 1 PR for HomeScreenFragment test coverage
- Move few logics to ViewModel as Fragment is not a correct place to do that stuff

- [x] Text Matching should be from string file

@shobhitagarwal1612  @sufyanAbbasi  PTAL?

Screenshot 2024-07-29 at 1 01 36 PM

### CODE REVIEW

File: ground/src/main/java/com/google/android/ground/ui/home/HomeScreenViewModel.kt [hunk 1]
```diff
@@ -42,7 +43,8 @@ internal constructor(
   private val navigator: Navigator,
   private val offlineAreaRepository: OfflineAreaRepository,
   private val submissionRepository: SubmissionRepository,
-  private val surveyRepository: SurveyRepository,
+  val surveyRepository: SurveyRepository,
+  val userRepository: UserRepository,
```
Comment (shobhitagarwal1612) [inline ground/src/main/java/com/google/android/ground/ui/home/HomeScreenViewModel.kt:47]: Let's keep these as private

**Review by shobhitagarwal1612 — CHANGES_REQUESTED** [review]

File: ground/src/main/java/com/google/android/ground/ui/home/HomeScreenViewModel.kt (hunk 1 above)
Comment (anandwana001) [reply ground/src/main/java/com/google/android/ground/ui/home/HomeScreenViewModel.kt:47]: These are getting used in HomeScreenFragment, so we cannot make them private

File: ground/src/main/java/com/google/android/ground/ui/home/HomeScreenFragment.kt [hunk 2]
```diff
@@ -58,10 +56,7 @@ class HomeScreenFragment :
   // TODO: It's not obvious which locations of interest are in HomeScreen vs MapContainer;
   //  make this more intuitive.
 
-  @Inject lateinit var locationOfInterestRepository: LocationOfInterestRepository
   @Inject lateinit var userRepository: UserRepository
```
[the remark this answers is not in the document]
Comment (anandwana001) [reply ground/src/main/java/com/google/android/ground/ui/home/HomeScreenFragment.kt:59]: Yes, planned to take this in next PR

**Pull request merged**